core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -840,7 +899,7 @@ protected void doRun() {
                     performOnReplica(shard);
                 }
                 // send operation to relocating shard
-                if (shard.relocating()) {
+                if (shard.relocating() && shard.relocatingNodeId().equals(nodes.localNodeId()) == false) {
```

Comment:
can we also change the comment to say that the local shard can be a relocation target of the primary ?